observer/probers/crl/crl_conf.go

```diff
@@ -0,0 +1,152 @@
+package probers
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/letsencrypt/boulder/observer/probers"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	nextUpdateName = "obs_crl_next_update"
+	thisUpdateName = "obs_crl_this_update"
+	certCountName  = "obs_crl_revoked_cert_count"
+)
+
+// CRLConf is exported to receive YAML configuration
+type CRLConf struct {
+	URL string `yaml:"url"`
+}
+
+// Kind returns a name that uniquely identifies the `Kind` of `Configurer`.
+func (c CRLConf) Kind() string {
+	return "CRL"
+}
+
+// UnmarshalSettings constructs a CRLConf object from YAML as bytes.
+func (c CRLConf) UnmarshalSettings(settings []byte) (probers.Configurer, error) {
+	var conf CRLConf
+	err := yaml.Unmarshal(settings, &conf)
+	if err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+func (c CRLConf) validateURL() error {
+	url, err := url.Parse(c.URL)
+	if err != nil {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, expected a valid url", c.URL)
+	}
+	if url.Scheme == "" {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, missing scheme", c.URL)
+	}
+	return nil
+}
+
+func (c CRLConf) validateCollectors(colls map[string]prometheus.Collector) (*prometheus.GaugeVec, *prometheus.GaugeVec, *prometheus.GaugeVec, error) {
+	if colls == nil {
+		message := "crl prober defines metrics but received nil collector map"
+		return nil, nil, nil, errors.New(message)
+	}
```

Comment:
I don't believe that this is necessary.

```go
import (
	"fmt"
	"log"
)

func main() {
	var foo map[string]string
	for k, v := range foo {
		if k == "bar" || v == "baz" {
			log.Panic()
		}
	}
	fmt.Printf("foo == nil is %v\n", foo == nil)
}
```

I get:
```
foo == nil is true
```
https://go.dev/play/p/LzPXpD7H55X

Reply:
wow, it looks like you can even attempt to read from a nil map and it'll just give you the zero value. golang is kinda crazy... anyways, done!